lec09-modeling-r.Rmd

```diff
@@ -1,4 +1,615 @@
+pp_flowField <- flowField(predator_prey, x.lim = c(0,10), y.lim = c(0,10),
+                          parameters = params,
+                          points = 15, # this is the density of grid points on which to plot arrows
+                          system = 'two.dim', # 'two.dim' is default
+                          add = FALSE)
+
+# add trajectories
+pp_trajectory <- trajectory(predator_prey, 
+                            # y0 is a matrix where each row is pairs of (X, Y) 
+                            y0 = initial_conditions, 
+                            t.end = 30, # how far in time to calculate the trajectories
+                            parameters = params, system = "two.dim")
+
+pp_nullclines <- nullclines(predator_prey, 
+           x.lim = c(0, 10), # axis limits
+           y.lim = c(0, 10),
+           # y0 is a matrix where each row is pairs of (X, Y) 
+           y0 = initial_conditions, 
+           t.end = 30, # how far in time to calculate the trajectories 
+           parameters = params, system = "two.dim")
+```
+
+The vector field tells us graphically that there is one stable fixed point at ($x = 1, y = 5$) and 
+one unstable fixed point at ($x = 0, y = 0$).
+
+#### Challenge (to think about)
+
+Is there a minimum number of null clines a system must have? Are the null clines
+guarranteed to intersect? Are null clines always straight lines?  
+
+```{r, eval=FALSE, echo=FALSE}
+
+# Consider $dx/dt = 3$. Maybe this isn't truly a differential equation, but $dx/dt$ can never be 0.
+
+# If each differential equation depends on at least one of the system variables 
+# (and the equations are different), there will always be at minimum two null clines. 
+
+# The null clines are not guarranteed to intersect: it's possible to not have a fixed point. 
+# Example: dx/dt = y, dy/dt = y + 2
+
+```
+
+#### Challenge
+
+Find the fixed point(s) and null clines analytically for the following two-dimensional 
+system. Verify your work by plotting the null clines. 
+
+
+## Quantitative stability analysis
+
+We've now found fixed points analytically, but up until now we've been analyzing the stability 
+of the fixed points by constructing a phase portrait and looking at it. This is a perfectly good
+way to analyze stability, but we can do more: we can find out a whole lot with the quantitative
+stability analysis we're about to do:
+
+- Whether a fixed point will be stable or unstable
+- How the system will behave near a fixed point (are there oscillations, etc.)
+- How the stability and type of a fixed point depend on the model parameters
+    
+The last reason in the list is in many ways the best reason. As we will see, in some models the
+number and/or stability of the fixed points depends on the system parameters, and quantitatively
+analyzing stability will show us this.
+
+To analyze stability, we **linearize** our model: we assume the system is linear close to the fixed
```

Comment:
Is it important to decide on *how* close to the fix point? I guess the curves here are quite smooth, but say we actually look at a mountain pass, then there would be many small bumps along the way but the overall trajectory would still be done towards the valley, are there rules for how to chose the step size to capture the overall direction? Maybe worth mentioning something about it without going into details?

Reply:
Yeah good point. In quantitative analysis you don't have to decide how close is close enough, because you're calculating the slope exactly at the fixed point.